packages/backend/src/queryBuilder.ts

```diff
@@ -118,6 +130,17 @@ export const buildQuery = ({
     const dimensionSelects = dimensions.map((field) => {
         const alias = field;
         const dimension = getDimensionFromId(field, explore);
+        // Throw error if user does not have the right requiredAttribute for this dimension
+        if (dimension.requiredAttributes)
```

Comment:
Users can bypass this by using the dimension in metric or table calculations.

Reply:
You can't use TC without having dimensions on the query 
![Screenshot from 2023-07-31 10-32-05](https://github.com/lightdash/lightdash/assets/1983672/ab84d648-53ea-46e0-8b9c-27bca4162a4d)


You can use custom metrics, but that is already included 
![image](https://github.com/lightdash/lightdash/assets/1983672/127bb10f-b473-4842-b9c0-a755a50d4e99)

Good comment though, I had to test this to make sure it works.

Reply:
I guess this is not a problem for viewers, 
developers and admins can already have access through sql runner. 
So this might only affect editors

@owlas should we limit those too ?

Reply:
You can't get a TC without referencing the dimension first 

![Uploading Screenshot from 2023-07-31 10-32-05.png…]()

But it is a problem with custom metrics :eyes:

Reply:
Fixed

![Screenshot from 2023-07-31 11-12-17](https://github.com/lightdash/lightdash/assets/1983672/71d8cc42-ba3a-4135-b4c3-98915fa5e433)


![Screenshot from 2023-07-31 11-12-54](https://github.com/lightdash/lightdash/assets/1983672/5ba53ca1-3748-4088-93b6-b13ec8bb9aec)

Reply:
@rephus and @ZeRego could we add the auth check to the `getDimensionById` function which is always used to convert a reference into a dimension. We already have a "dimension does not exists" error. We can add to that an auth error depending on attributes. I think this would catch most (probably all) use cases